tests/memory_checker/test_memory_checker.py

```diff
@@ -131,26 +208,74 @@ def consume_memory(duthost, container_name, vm_workers):
+
+    Args:
+        duthost: The AnsibleHost object of DuT.
+        container_name: Name of container.
+
+    Returns:
+        None.
+    """
+    status_result = duthost.critical_process_status(container_name)
+    if status_result["status"] is False or len(status_result["exited_critical_process"]) > 0:
+        return False
+
+    return True
+
+
+def postcheck_critical_processes(duthost, container_name):
+    """Checks whether the critical processes are running after container was restarted.
+
+    Args:
+        duthost: The AnsibleHost object of DuT.
+        container_name: Name of container.
+
+    Returns:
+        None.
+    """
+    logger.info("Checking the running status of critical processes in '{}' container ..."
+                .format(container_name))
+    is_succeeded = wait_until(CONTAINER_RESTART_THRESHOLD_SECS, CONTAINER_CHECK_INTERVAL_SECS, 0,
+                              check_critical_processes, duthost, container_name)
+    if not is_succeeded:
+        pytest.fail("Not all critical processes in '{}' container are running!"
+                    .format(container_name))
+    logger.info("All critical processes in '{}' container are running.".format(container_name))
+
+
+def consumes_memory_and_checks_container_restart(duthost, container_name, vm_workers):
     """Invokes the 'stress' utility to consume memory more than the threshold asynchronously
-    and checks whether the container can be stopped and restarted. Loganalyzer was leveraged
+    and checks whether the container can be stopped and restarted. Loganalyzer is leveraged
     to check whether the log messages related to container stopped were generated.
 
     Args:
         duthost: The AnsibleHost object of DuT.
-        container_name: Name of container.
+        container_name: A string represents the name of container.
         vm_workers: Number of workers which does the spinning on malloc()/free()
           to consume memory.
 
     Returns:
         None.
-
     """
+    expected_alerting_messages = []
+    expected_alerting_messages.append(".*restart_service.*Restarting service 'telemetry'.*")
+    expected_alerting_messages.append(".*Stopping Telemetry container.*")
+    expected_alerting_messages.append(".*Stopped Telemetry container.*")
+    expected_alerting_messages.append(".*Starting Telemetry container.*")
+    expected_alerting_messages.append(".*Started Telemetry container.*")
+
+    loganalyzer = LogAnalyzer(ansible_host=duthost, marker_prefix="container_restart_due_to_memory")
+    loganalyzer.expect_regex = []
+    loganalyzer.expect_regex.extend(expected_alerting_messages)
+    marker = loganalyzer.init()
+
     thread_pool = ThreadPool()
     thread_pool.apply_async(consume_memory, (duthost, container_name, vm_workers))
 
-    logger.info("Sleep 100 seconds to wait for the alerting messages from syslog...")
-    time.sleep(100)
+    logger.info("Sleep 130 seconds to wait for the alerting messages from syslog ...")
```

Comment:
>130 [](http://example.com/codeflow?start=23&length=3)

How do you get the magic number of seconds?
Is it platform dependent? If yes, what is the guideline to calculate for a new platform?

Reply:
Currently polling interval (cycle) of Monit is 60 seconds. Since in Monit configuration file of `telemetry`, it will be restarted if memory usage is larger than `400MB` for 1 time within `2 cycles`. `2 cycles` will be 120 seconds. That's why at here number `130` is used.

This number is not platform dependent.